ara/ui/views.py

```diff
@@ -53,6 +53,68 @@ def get(self, request, *args, **kwargs):
         # fmt: on
 
 
+class HostIndex(generics.RetrieveAPIView):
+    """
+    Returns the latest playbook result for each host (or all playbook results for every hosts)
+    """
+
+    renderer_classes = [TemplateHTMLRenderer]
+    pagination_class = LimitOffsetPaginationWithLinks
+    template_name = "host_index.html"
+
+    def get(self, request, *args, **kwargs):
+        search_form = forms.HostSearchForm(request.GET)
+
+        # Default is DistinctHost (by not requiring "?latest=true") but accept false to
+        # return all hosts
+        if "latest" in request.GET and request.GET["latest"] == "false":
+            queryset = models.Host.objects.all()
+            serializer_type = "DetailedHostSerializer"
+            filter_type = "HostFilter"
+            # TODO: Is there a cleaner way ? Doing this logic in the template seemed complicated.
+            checkbox_status = "checked"
+        else:
+            queryset = models.DistinctHost.objects.all()
+            serializer_type = "DetailedDistinctHostSerializer"
+            filter_type = "DistinctHostFilter"
+            checkbox_status = ""
+
+        # Sort by updated by default so we have the most recently updated at the top
+        order = "-updated"
+        if "order" in request.GET:
+            order = request.GET["order"]
+
+        query = getattr(filters, filter_type)(request.GET, queryset=queryset)
+        page = self.paginate_queryset(query.qs.all().order_by(order))
```

Comment:
I think there might be a bug here.

It is ordering by ``-updated`` which is the ``updated`` field of the latesthost, *NOT* of the actual host.
This would be fine off of a fresh database since the "updated" field would continuously get updated.

However, because we are not supplying the "created" and "updated" timestamps when running the SQL migration, they are set to "now" instead of "whenever it was for that host" so sorting by date on a database that has run migrations leads to results that are not completely wrong but a bit weird.

Reply:
Visualized here after running ``ara-manage migrate``:
![Screenshot from 2021-07-26 18-21-47](https://user-images.githubusercontent.com/1291204/127067279-3601cb58-6883-4ea1-b942-bdacbd7e417a.png)

I tried to force the ``created`` and ``updated`` fields to match the child host inside the sql migration and the best I've got was the created field because, well, doing an update updates the updated field to "now" :P

![Screenshot from 2021-07-26 18-28-45](https://user-images.githubusercontent.com/1291204/127067387-69769f57-e149-4351-8b72-3ae84745af09.png)

This is tricky. I'll think about it.

Reply:
Additional note: this is only an issue for the view with the latest playbook per host because the updated field is not accurate. When enabling the toggle to view all hosts, the data is ordered correctly.

Reply:
This makes me wonder: should we maybe remove the ``auto_now`` and ``auto_now_add`` on the model ?
https://github.com/ansible-community/ara/blob/7c5988ae7d0c1c1779b2437ff7c52e04e265c52d/ara/api/models.py#L256-L258

For example, if a host gets deleted, the latesthost reference will be updated and thus bumping the updated field but it doesn't mean the host that has taken it's place has been updated recently.

Sorting by created doesn't exactly work either because the latesthost will probably always have the same created date (unless it gets deleted and re-created).

If we remove the auto_now and auto_now_add, we would be "free" to put the dates we want in the sql migration as well as in the creation/deletion management:
https://github.com/ansible-community/ara/blob/7c5988ae7d0c1c1779b2437ff7c52e04e265c52d/ara/api/views.py#L108-L113

https://github.com/ansible-community/ara/blob/7c5988ae7d0c1c1779b2437ff7c52e04e265c52d/ara/api/serializers.py#L341

Reply:
in my opinion the updated and created fields of the DistinctHost/ LatestHost objects are describing when this object was firstly created and when it was updated. It should NOT equals to the updated and created field of the related host object.

Reply:
> Additional note: this is only an issue for the view with the latest playbook per host because the updated field is not accurate. When enabling the toggle to view all hosts, the data is ordered correctly.

I think the cleanest solution is if you are using the updated field of the related host object in the view and not the updated field of the DistinctHost/LatestHost object itself.
Probably more difficult, but a clean solution ;)

Reply:
@hille721 fair enough -- if the intent is to sort by the updated field of the child hosts, is there a way to do it without relying on the updated field of the latesthost ?

Reply:
this should work:

```
.order_by('latest__updated')
```

Reply:
I didn't know you could sort by nested fields like that, I will try it !

Reply:
It works !

I've sent a PR to fix it: https://github.com/ansible-community/ara/pull/300. Thank you :)